Watchman.Discord/Areas/Statistics/Controllers/StatisticsController.cs

```diff
@@ -39,28 +29,25 @@ public StatisticsController()
             this._chartsService = new ChartsService();
         }
 
-        //TODO informations about author, channel and server should be set in middleware 
-        //so that MessageContext or UserContext & ChannelContext & ServerContext objects should be added
-
         [ReadAlways]
-        public void SaveMessage(SocketMessage message)
+        public void SaveMessage(string message, Dictionary<string, IDiscordContext> contexts)
         {
-            //todo wrzucić budowanie tego obiektu w osobną klase -> MessageInformationBuilder
+            var userContext = (UserContext) contexts[nameof(UserContext)];
+            var channelContext = (ChannelContext) contexts[nameof(ChannelContext)];
 
-            //probably should be structures, not classes
             var author = new MessageInformationAuthor
```

Comment:
To bym wrzucił do osobnej klasy, jakiś MessageInformationBuilder
żeby nie zajmowało to tyle miejsca w kontrolerze